service/iot-service-client/src/main/java/com/microsoft/azure/sdk/iot/service/digitaltwin/DigitalTwinAsyncClient.java

```diff
@@ -44,18 +45,18 @@
      * @param connectionString The IoTHub connection string
      */
     @Builder(builderMethodName = "buildFromConnectionString", builderClassName = "FromConnectionStringBuilder")
-    public DigitalTwinAsyncClient(@NonNull String connectionString) {
-        ServiceConnectionString serviceConnectionString = ServiceConnectionStringParser.parseConnectionString(connectionString);
-        SasTokenProvider sasTokenProvider = serviceConnectionString.createSasTokenProvider();
-        String httpsEndpoint = serviceConnectionString.getHttpsEndpoint();
+    public DigitalTwinAsyncClient(@NonNull String connectionString) throws IOException {
+        IotHubConnectionString iotHubConnectionString = IotHubConnectionStringBuilder.createConnectionString(connectionString);
+        IotHubServiceSasToken iotHubServiceSasToken = new IotHubServiceSasToken(iotHubConnectionString);
```

Comment:
do we get token caching and refresh?

Reply:
Good point. I will check.

Reply:
If you have context about this, let me know.

Reply:
Do you know if we had that in the copied code?

Reply:
Currently we have an expiry of 1 year for the SAS token so this implementation should be sufficient. We should revisit the logic of this in a separate task. I don't think maintaining code separately for DigitalTwinClient is the best option.

Reply:
I'm curious about this too. Can we talk about it in parking lot or in a meeting?

Reply:
yes sure. I wanted to discuss with Abhipsa and the pnp team before bringing it to standup.